File: Class Code/Sorting/4_Merge_Sort/2_Program.c
//Kth Smallest Number Find after Sorting Array :

#include<stdio.h>

int merge(int arr[],int start,int mid,int end){
	
	int ele1 = mid - start + 1;
	int ele2 = end - mid;

	int arr1[ele1],arr2[ele2];

	for(int i = 0;i < ele1;i++){
		
		arr1[i] = arr[start + i];
	}
	for(int i = 0;i < ele2;i++){
		arr2[i] = arr[mid + 1 + i];
	}

	int itr1 = 0,itr2 = 0,itr3 = start;

	while(itr1 < ele1 && itr2 < ele2){
		if(arr1[itr1] < arr2[itr2]){
			arr[itr3] = arr1[itr1];
			itr1++;
		}else{
			arr[itr3] = arr2[itr2];
			itr2++;
		}
		itr3++;
	}
	while(itr1 < ele1){
		arr[itr3] = arr1[itr1];
		itr3++;
		itr1++;
	}
	while(itr2 < ele2){
		arr[itr3] = arr2[itr2];
		itr2++;
		itr3++;
	}

}
void merge_Sort(int arr[],int start,int end){
	
	if(start < end){
		
		int mid = (start + end)/2;

		merge_Sort(arr,start,mid);
		merge_Sort(arr,mid + 1,end);

		merge(arr,start,mid,end);
	}

}

void main(){
	int size;
	printf("Enter the size of Array : ");
	scanf("%d",&size);

	int arr[size];
	printf("Enter the element in Array : ");
	for(int i = 0;i < size;i++){
		
		scanf("%d",&arr[i]);
	}
	int k;
	printf("Kth small number :");
	scanf("%d",&k);

	merge_Sort(arr,0,size-1);
	printf("Sorted Array : \n");
	for(int i = 0;i < size;i++){
		
		printf("%d ",arr[i]);
	}
	printf("\n");
}

File: Class Code/Searching/Platform Codes/2_Program.java


class Solution {
    static int binarysearch(int arr[], int n, int k) {
        int start = 0;
        int end = n - 1;
        while(start < end){

            int mid = start + (end - start)/2;
		
	    System.out.println(mid);
            if(arr[mid] == k){

                return mid;
            }else if(arr[mid] > k){

                end = mid - 1;
            }else if(arr[mid] < k){

                start  = mid + 1;
            }
        }
        return  -1;
    }
    public static void main(String[] args){

	int arr[] = {1,2 ,3 ,4 ,5 ,6 ,8 ,9 ,10, 14, 16, 19, 22, 23, 25, 26, 27, 29, 31, 34, 35, 36, 38, 39, 40, 45, 46, 48, 50 ,51, 52, 57, 59, 60, 61, 63, 67, 68, 69, 71, 75, 76, 77, 79, 81, 82, 83, 86, 87, 88, 90, 92, 93, 94, 95, 96, 98, 99, 100};

	int ret = binarysearch(arr,59,93);
	System.out.println(ret);
    }
}
